src/utils/miscellaneous.ts

```diff
@@ -10,5 +10,13 @@ export function getToken<T extends TokenDetails>(
   value: string | undefined = '',
   tokens: T[] | undefined | null,
 ): T | undefined {
-  return (tokens || []).find(token => token[key] === value.toUpperCase())
+  const valueUppercase = value.toUpperCase()
+  return (tokens || []).find(token => {
+    const value = token[key]
+    if (value) {
+      return value.toString().toUpperCase() === valueUppercase
+    } else {
+      return false
```

Comment:
because this is generic it's not complaining but this return should be undefined. actually you can just leave out the `else` as it'll auto return `undefined` if not found.

Reply:
I think the doc says that it should return a `truthy` value when you find the element and a `falsy` value if the current element is not what you look for.

It's true that `undefined` is `falsy` and I can remove the else, but I find it more declarative to use a function that returns always a boolean. `false` it's also a good `falsy` value :) 

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find

Reply:
In that case, update the function return value to include `false`

Reply:
ok, then type the function return as `boolean`

Reply:
The linter didn't enforce me to do that, but if you think it's more clear, I'll add it explicitly